lib/level_db.rb

```diff
@@ -113,6 +113,24 @@ def each(options={}, &block)
     def snapshot
       Snapshot.new(@db)
     end
+
+    def compact_range(options={})
+      from = options[:from]
+      to = options[:to]
+      @db.compact_range(encode_key(from), encode_key(to))
+    end
+
+    def full_compaction
+      sstables = @db.get_property(SSTABLES_PROPERTY)
+      pre_compaction_sstables = nil
+      until pre_compaction_sstables == sstables
+        pre_compaction_sstables = sstables
+        compact_range
+        sstables = @db.get_property(SSTABLES_PROPERTY)
+      end
```

Comment:
Looking at the leveldb source code, [the maximum number of levels seems to be 7](https://github.com/google/leveldb/blob/77948e7eec0613fb5cbecb7e320b9498607030b5/db/dbformat.h#L22). Also, [the format of returned string doesn't strike me as easily parsed/sorted](https://github.com/google/leveldb/blob/77948e7eec0613fb5cbecb7e320b9498607030b5/db/version_set.cc#L565-L589).